draft-ietf-httpbis-http2bis.xml

```diff
@@ -2934,16 +2934,33 @@ cookie: e=f
                 pseudo-header field to convey authority information, unless there is no authority
                 information to convey (in which case it MUST NOT generate :authority).
               </t>
+              <t>
+                Clients MUST NOT generate a request with a <tt>Host</tt> header field that differs
+                from the <tt>:authority</tt> pseudo-header field, when compared byte-for-byte.  A
+                server SHOULD treat a request as malformed if it contains a <tt>Host</tt> header
+                field that identifies a different entity to the <tt>:authority</tt> pseudo-header
+                field.  The values of fields need to be normalized to compare them (see <xref
```

Comment:
Does "of fields" apply to all fields, all pseudo fields or just :authority fields?

Does the normalization "need" apply to the byte-for-byte comparison above ie. do MUST they only be byte-for-byte identical after normalization?
  
Is a "need" a MUST or a SHOULD?

Reply:
In this instance, I think "need" is a statement of fact -- these are hostnames, therefore not case-sensitive, therefore they need to be normalized to perform a useful comparison.  No need for RFC2119 language here.  However, that implies that "when compared byte-for-byte" above is wrong.

Reply:
We're placing a stronger requirement on the sender than the receiver.  The receiver requirements are more lax because we want to allow for folks like @icing who might want to avoid extra work to do so (either because they are happy that their h2 implementation is otherwise OK and don't want to open the patient to add extra checks that will mostly just slow them down).  We don't need to see every server doing this rejection, just a few.  Clients will learn not to put different values in.

That said, I think we can say "identical" in the first part without resorting to "byte-for-byte".  "byte-for-byte" is just the most obvious and easiest way to ensure that the values are the same.

Reply:
> ...we want to allow for folks like @icing who might want to avoid extra work to do so...

That's very kind and heart-warming.😌